### PR TITLE
258 Splash Screen

### LINKED ISSUE #258: UI Feature - Splash Screen
As a user, I would like to get feedback from the application on the operations performed and user actions.

Acceptance Criteria:
- [x] Mock-ups have been created and approved by the team
- [x] Mock-ups have been updated on Confluence
- [x] Opening the application displays an app opening screen before getting to the main page (asset thumbnails)

Question:
How long does the screen gets displayed for?

### PR DESCRIPTION
**Pull request by najimgh**
Added a Splash Screen which will be shown before the Assets Page.

This resolves #258

### CODE REVIEW

File: JavaFXUserInterface/src/main/java/Controllers/SplashScreenController.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    @Override
+    public void initialize(URL url, ResourceBundle resourceBundle) {
+        new SplashScreen().start();
+    }
+
+    class SplashScreen extends Thread {
+        @Override
+        public void run() {
+            try {
+                Thread.sleep(5500); //Displays the SplashScreen for 5.5 seconds
```
Comment (PaulMicu) [inline JavaFXUserInterface/src/main/java/Controllers/SplashScreenController.java]: I dont think there should be a hardcoded amount of time the splash screen is displayed, it should be displayed for as long as the app takes to boot up, no longer, no shorter

**Review by PaulMicu — CHANGES_REQUESTED** [review]

File: JavaFXUserInterface/src/main/resources/SplashScreen.fxml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            </ImageView>
+         </children>
+      </HBox>
+      <HBox alignment="CENTER">
+         <children>
+            <Label text="M i n e r v a" textAlignment="CENTER" textFill="WHITE">
+               <font>
+                  <Font name="Futura Medium" size="55.0"/>
+               </font>
+            </Label>
```
Comment (jeffbuonamici) [inline JavaFXUserInterface/src/main/resources/SplashScreen.fxml]: Text Alignment, Text fill, font family, and font size should just be in the css

File: JavaFXUserInterface/src/main/resources/css/style.css [hunk 3]
```diff
@@ -67,4 +67,9 @@
 .input-error {
     -fx-border-color: red;
     -fx-border-width: 2px;
+}
+
+.splash-screen-bg {
+    -fx-background-color: rgb(22, 53, 96);
+    -fx-background-color: linear-gradient(to top right, rgba(22, 53, 96, 1) 0%, rgba(63, 143, 204, 1) 85%);;
```
Comment (jeffbuonamici) [inline JavaFXUserInterface/src/main/resources/css/style.css]: Is there a reason why there's two background-color attributes?

**Review by jeffbuonamici — CHANGES_REQUESTED** [review]

File: JavaFXUserInterface/src/main/resources/SplashScreen.fxml (hunk 2 above)
Comment (najimgh) [reply JavaFXUserInterface/src/main/resources/SplashScreen.fxml]: Fixed

File: JavaFXUserInterface/src/main/resources/css/style.css (hunk 3 above)
Comment (najimgh) [reply JavaFXUserInterface/src/main/resources/css/style.css]: The extra background-color attribute has been removed.

File: JavaFXUserInterface/src/main/java/Controllers/SplashScreenController.java (hunk 1 above)
Comment (najimgh) [reply JavaFXUserInterface/src/main/java/Controllers/SplashScreenController.java]: Splash Screen is now implemented using a preloader with a fade transition. The Splash Screen is displayed as long as all of the assets are not loaded completely.

**Review by jeffbuonamici — APPROVED** [review]
My concerns were addressed. Approved.

**Review by PaulMicu — APPROVED** [review]

**Pull request merged**